### DISCUSSION TITLE
v0.29.0

### DISCUSSION THREAD

**Discussion by borg323** [LeelaChessZero/lc0#1812] [General]
In this release:
* New metal backend for apple systems. This is now the default backend for macos builds.
* New onnx-dml backend to use DirectML under windows, has better net compatibility than dx12 and is faster than opencl. See the README for use instructions, a separate download of the DirectML dll is required.
* Full attention policy support in cuda, cudnn, metal, onnx, blas, dnnl, and eigen backends.
* Partial attention policy support in onednn backend (good enough for T79).
* Non multigather (legacy) search code and `--multigather` option are removed.
* Now the onnx backends can use fp16 when running with a network file (not with .onnx model files). This is the default for onnx-cuda and onnx-dml, can be switched on or off with by setting the `fp16` backend option to `true` or `false` respectively.
* The onednn package comes with the latest dnnl compiled to allow running on an intel gpu by adding `gpu=0` to the backend options.
* The default net is now 791556 for most backends except opencl and dx12 that get 753723 (as they lack attention policy support).
* Support for using pgn book with long lines in training: selfplay can start at a random point in the book.
* New "simple" time manager.
* Support for double Fischer random chess (dfrc).
* Added TC-dependent output to the backendbench assistant.
* Starting with this version, the check backend compares policy for valid moves after softmax.
* The onnx backend now allows selecting gpu to use.
* Improved error messages for unsupported network files.
* Some assorted fixes and code cleanups.

<hr /><em>This discussion was created from the release <a href='https://github.com/LeelaChessZero/lc0/releases/tag/v0.29.0'>v0.29.0</a>.</em>

**Comment by Avansh1234** [comment]
What

**Comment by jerson-francisco** [comment]
how to download leela zero for pc

**Comment by jerson-francisco** [comment]
pls give me
